applications/ParticleMechanicsApplication/python_scripts/apply_mpm_particle_dirichlet_condition_process.py

```diff
@@ -90,9 +116,53 @@ def __init__(self, Model, settings ):
                 condition.Set(KratosMultiphysics.MODIFIED, self.modified_normal)
                 condition.SetValue(KratosParticle.PARTICLES_PER_CONDITION, self.particles_per_condition)
                 condition.SetValue(KratosParticle.MPC_IS_NEUMANN, self.is_neumann_boundary)
-                condition.SetValue(KratosParticle.PENALTY_FACTOR, self.penalty_factor)
-                condition.SetValue(self.variable, self.vector)
+                condition.SetValue(KratosParticle.MPC_BOUNDARY_CONDITION_TYPE, self.boundary_condition_type)
+
+                ### Set necessary essential BC variables
+                if self.boundary_condition_type==1:
+                    condition.SetValue(KratosParticle.PENALTY_FACTOR, self.penalty_factor)
```

Comment:
since the value is always the same maybe makes more sense to put it into the processinfo what do you think?

Reply:
you mean the Penalty_Factor? At the moment it is constant during the simulation but, as it is a problem dependent parameter, it could be possible/necessary to change it during the simulation

Reply:
I mean if it the same for all conditions then I would put it to the ProcessInfo

Reply:
mmmh....usually it will be the same for all conditions but not necessarily. Does it make a big difference concerning memory and computation time?

Reply:
if there is the possibility that it varies I would leave it

Compared to the rest the additional mem consumption is not important I would say

Reply:
ok